server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -265,122 +260,124 @@ public ContentAccessCertificate getCertificate(Consumer consumer)
-            Date contentUpdate = owner.getLastContentUpdate();
-
-            if (expiration.before(now) || !contentUpdate.before(existing.getUpdated()) ||
-                oeca == null || !contentUpdate.before(oeca.getUpdated())) {
 
-                consumer.setContentAccessCert(null);
-                this.contentAccessCertificateCurator.delete(existing);
-
-                regenerate = true;
-            }
+        ContentAccessCertificate result;
+        if (existing == null) {
+            result = createNewScaCertificate(consumer, owner);
+        }
+        else {
+            result = updateScaCertificate(consumer, owner, existing);
         }
 
-        if (existing == null || regenerate) {
-            log.info("Generating new SCA certificate for organization \"{}\"...", owner.getKey());
-
-            Calendar cal = Calendar.getInstance();
-            cal.add(Calendar.HOUR, -1);
-            Date startDate = cal.getTime();
-            cal.add(Calendar.YEAR, 1);
-            Date endDate = cal.getTime();
+        return wrap(result);
+    }
 
-            CertificateSerial serial = new CertificateSerial(endDate);
-            // We need the sequence generated id before we create the Certificate,
-            // otherwise we could have used cascading create
-            serialCurator.create(serial);
+    private ContentAccessCertificate createNewScaCertificate(Consumer consumer, Owner owner)
+        throws IOException, GeneralSecurityException {
+        log.info("Generating new SCA certificate for consumer: \"{}\"", consumer.getUuid());
+        Validity oneYearValidity = Validity.oneYear();
+        CertificateSerial serial = createSerial(oneYearValidity);
+
+        KeyPair keyPair = this.keyPairCurator.getConsumerKeyPair(consumer);
+        byte[] pemEncodedKeyPair = pki.getPemEncoded(keyPair.getPrivate());
+
+        ContentAccessCertificate existing = new ContentAccessCertificate();
+        existing.setSerial(serial);
+        existing.setKeyAsBytes(pemEncodedKeyPair);
+        existing.setConsumer(consumer);
+
+        existing.setCert(createX509Cert(consumer, owner, serial, keyPair, oneYearValidity));
+        existing.setContent(this.createPayloadAndSignature(owner, consumer));
+        ContentAccessCertificate savedCert = this.contentAccessCertificateCurator.create(existing);
+        consumer.setContentAccessCert(savedCert);
+        this.consumerCurator.merge(consumer);
+        return savedCert;
+    }
 
+    private ContentAccessCertificate updateScaCertificate(Consumer consumer, Owner owner,
+        ContentAccessCertificate existing) throws GeneralSecurityException, IOException {
+        Date now = new Date();
+        Date expiration = existing.getSerial().getExpiration();
+        Date contentUpdate = owner.getLastContentUpdate();
+        // Checks need to be done before any updates!
+        boolean shouldUpdateContent = !contentUpdate.before(existing.getUpdated());
```

Comment:
not sure why this needs to be checked here, instead of right before `if (shouldUpdateContent) {`?